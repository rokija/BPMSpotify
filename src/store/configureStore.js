import {createStore, compose, applyMiddleware} from 'redux';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import thunk from 'redux-thunk';
import createHistory from 'history/createBrowserHistory';
import { routerMiddleware } from 'react-router-redux';
import rootReducer from '../reducers';
export const history = createHistory();
export function configureStoreProd(initialState) {
  const reactRouterMiddleware = routerMiddleware(history);
  const middlewares = [
    thunk,
    reactRouterMiddleware,
  ];

  return createStore(rootReducer, initialState, compose(
    applyMiddleware(...middlewares)
    )
  );
}

function configureStoreDev(initialState) {
  const reactRouterMiddleware = routerMiddleware(history);
  const middlewares = [
    reduxImmutableStateInvariant(),
    thunk,
    reactRouterMiddleware,
  ];

  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  const store = createStore(rootReducer, initialState, composeEnhancers(
    applyMiddleware(...middlewares)
    )
  );

  /* istanbul ignore next */
  /* can't test hot reloading */
  if (module.hot) {
    module.hot.accept('../reducers', () => {
      const nextReducer = require('../reducers').default;
      store.replaceReducer(nextReducer);
    });
  }

  return store;
}

/* istanbul ignore next */
/* can't simulate production mode for test environment*/
const configureStore = process.env.NODE_ENV === 'production' ? configureStoreProd : configureStoreDev;

export default configureStore;
