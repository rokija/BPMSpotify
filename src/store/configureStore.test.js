import configureStore, { configureStoreProd } from './configureStore';

describe('configureStore', () => {
    it('returns initial State for configureStoreDev', () => {
        const store = configureStore();
        expect(store.getState()).toEqual({"audioFeatures": {}, "authReducer": {}, "routing": {"location": null}, "search": {}});
    });

    it('returns initial State for configureStoreProd', () => {
        const store = configureStoreProd();
        expect(store.getState()).toEqual({"audioFeatures": {}, "authReducer": {}, "routing": {"location": null}, "search": {}});
        expect(store.getState()).toMatchSnapshot();
    });
});