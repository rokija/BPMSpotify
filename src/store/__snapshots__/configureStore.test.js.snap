// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`configureStore returns initial State for configureStoreProd 1`] = `
Object {
  "audioFeatures": Object {},
  "authReducer": Object {},
  "routing": Object {
    "location": null,
  },
  "search": Object {},
}
`;
